shopinvader/views/partner_view.xml

```diff
@@ -35,30 +20,188 @@
+                            />
+                        <field name="display_validate_address" invisible="1" />
+                        <button
+                                class="btn btn-sm btn-warning"
+                                name="action_shopinvader_validate_customer"
+                                type="object"
+                                title="Validate user for Shopinvader"
+                                attrs="{'invisible':['|', ('has_shopinvader_user_active', '=', True), ('has_shopinvader_user', '=', False) ]}"
+                                icon="fa-user-o"
+                            >
+                          Shop: validate user
+                        </button>
+                        <div
+                                attrs="{'invisible':[('display_validate_address', '=', False)]}"
+                            >
+                          <button
+                                    class="btn btn-sm btn-success"
+                                    name="action_shopinvader_validate_address"
+                                    type="object"
+                                    title="Validate address for Shopinvader"
+                                    attrs="{'invisible':[('is_shopinvader_active', '=', True)]}"
+                                    icon="fa-address-card-o"
+                                >
+                            Shop: validate address
+                          </button>
+                          <button
+                                    class="btn btn-sm btn-warning"
+                                    name="action_shopinvader_validate_address"
+                                    type="object"
+                                    context="{'default_next_state': 'inactive'}"
+                                    title="Deactivate address for Shopinvader"
+                                    attrs="{'invisible':[('is_shopinvader_active', '=', False)]}"
+                                    icon="fa-address-card-o"
+                                >
+                            Shop: deactivate address
+                          </button>
+                        </div>
+                      </group>
+                    </group>
                 </page>
             </notebook>
-            <!-- add validation button to address kanban -->
-            <xpath
-                expr="//field[@name='child_ids']/kanban/field[@name='display_name']"
-                position="after"
-            >
-                <field name="shopinvader_enabled" />
+            <xpath expr="//field[@name='child_ids']" position="before">
+                <field name="has_shopinvader_user_to_validate" invisible="1" />
+                <field name="has_shopinvader_address_to_validate" invisible="1" />
+                <button
+                    class="btn btn-sm btn-success"
+                    name="action_shopinvader_validate_customer"
+                    context="{'validate_all': True}"
```

Comment:
This isn't used afaics, there's no special treatment in `_get_shopinvader_validate_wizard` and context gets lost here:

https://github.com/shopinvader/odoo-shopinvader/blob/4f581bf2742b1c120e510a03128fea7175735769/shopinvader/models/shopinvader_partner.py#L193-L198

Not sure what the intention is though, just pointing something to look at

Reply:
why do you say is not used? It used to change the behavior: if you are on the main partner you want to validate children

Reply:
It's only used in `action_shopinvader_validate_address`, but not here (`action_shopinvader_validate_customer`), AFAICS.

It ends up calling `action_shopinvader_validate`, which reads an action and returns it (context gets lost there)

Reply:
actually the ctx key is used by _get_shopinvader_validate_wizard to pick up the right records. Am I missing anything?

Reply:
Here? 

https://github.com/shopinvader/odoo-shopinvader/blob/4f581bf2742b1c120e510a03128fea7175735769/shopinvader/models/shopinvader_partner.py#L200-L202

Maybe you meant `_get_shopinvader_validate_address_wizard` ?
https://github.com/shopinvader/odoo-shopinvader/blob/4f581bf2742b1c120e510a03128fea7175735769/shopinvader/models/res_partner.py#L222-L233

Reply:
I'll double check later. NOTE: this stuff comes from the other PR that I've just refactored here https://github.com/shopinvader/odoo-shopinvader/pull/1022 -> let's move the conversation there :wink: